eZ/Publish/Core/FieldType/Image/ImageStorage/Gateway/DoctrineStorage.php

```diff
@@ -222,6 +252,20 @@ public function countImageReferences($uri)
         return (int) $statement->fetchColumn();
     }
 
+    public function countDistinctImages(): int
+    {
+        $selectQuery = $this->connection->createQueryBuilder();
+        $selectQuery
+            ->select($this->connection->getDatabasePlatform()->getCountExpression('contentobject_attribute_id'))
+            ->from($this->connection->quoteIdentifier(self::IMAGE_FILE_TABLE))
+            ->groupBy('contentobject_attribute_id', 'filepath')
+        ;
```

Comment:
Consider the following data:
```
id | contentobject_attribute_id | filepath 
----+----------------------------+----------
  1 |                          1 | abc
  2 |                          2 | abc
  3 |                          3 | def
```
That query for those data would return:
```
count 
-------
     1
     1
     1
```
However at the end you're fetching just first row / column. This does not make sense to me at all. What are you trying to achieve here?

Reply:
Depending on actions taken, you might end up with records looking like this:
![image](https://user-images.githubusercontent.com/38975612/120196053-b1eff580-c21f-11eb-9e8f-7f521c28dc2a.png)
There is no need to run the normalization command 6 times when there are only 3 unique images.
I want to count only unique images. I've just noticed that the image directory changes with every version and `content_object_attriibute_id` is included there as well. I guess I just overcomplicated things - I'll rework the query.